x-pack/plugins/lens/public/datatable_visualization/expression.tsx

```diff
@@ -127,14 +148,38 @@ export const getDatatableRenderer = (
   },
 });
 
-function DatatableComponent(props: DatatableProps & { formatFactory: FormatFactory }) {
+export function DatatableComponent(props: DatatableRenderProps) {
   const [firstTable] = Object.values(props.data.tables);
   const formatters: Record<string, ReturnType<FormatFactory>> = {};
 
   firstTable.columns.forEach(column => {
     formatters[column.id] = props.formatFactory(column.formatHint);
   });
 
+  const handleFilterClick = (field: string, value: unknown, index: number, negate = false) => {
+    const timeFieldName = negate
+      ? undefined
+      : firstTable.columns.find(col => col?.meta?.type === 'date_histogram')?.meta?.aggConfigParams
+          ?.field;
```

Comment:
This approach assumes that the column being clicked is the first date column. The table can have two or more date columns in any order, so the timeFieldName is not necessarily first. Try the following scenario:

1. Use the ecommerce sample data
2. Drag in products.created_on
3. Drag in order_date onto the "Drop a field here" new column
4. Filtering on the `order_date` is now going to create a filter in the filter bar instead of on the time picker